cmd/pbackend/statuser.go

```diff
@@ -264,22 +286,24 @@ func sendResults(ctx context.Context, batchSize int, tickDuration time.Duration)
 				messages = messages[:0]
 			}
 		case <-ticker.C:
+			logger := zerolog.Ctx(cancelCtx)
 			length := len(messages)
 			if length > 0 {
 				logger.Trace().Int("messages", length).Msgf("Sending %d source availability status messages (tick)", length)
-				err := kafka.Send(ctx, messages...)
+				err := kafka.Send(cancelCtx, messages...)
 				if err != nil {
 					logger.Warn().Err(err).Msg("Could not send source availability status messages (tick)")
 				}
 				messages = messages[:0]
 			}
-		case <-ctx.Done():
+		case <-cancelCtx.Done():
+			logger := zerolog.Ctx(cancelCtx)
 			ticker.Stop()
 			length := len(messages)
 
 			if length > 0 {
 				logger.Trace().Int("messages", length).Msgf("Sending %d source availability status messages (cancel)", length)
-				err := kafka.Send(ctx, messages...)
+				err := kafka.Send(cancelCtx, messages...)
 				if err != nil {
 					logger.Warn().Err(err).Msg("Could not send source availability status messages (cancel)")
 				}
```

Comment:
This was a mess of misusing context, now it has `cancelCtx` which is only used for context cancellation and where message is not available (e.g. shutting down the service). For everything else there is `messageCtx` which is copied to `ctx` variable so it is more natural to write and copy-paste code.